### PR TITLE
Added BuffTypes arguments.

### PR DESCRIPTION
**Pull request by yasinallahdev**
All calls to AddBuffHudVisual in the Subrepo now use a BuffType argument on creation. This Pull Request should not be merged until LeagueSandbox/GameServer#711 is merged.

### CODE REVIEW

File: Champions/Global/SummonerRevive.cs [hunk 1]
```diff
@@ -27,10 +28,9 @@ public void OnFinishCasting(Champion owner, Spell spell, AttackableUnit target)
             statMod.MoveSpeed.PercentBonus = 125.0f / 100.0f;
             owner.AddStatModifier(statMod);
             ApiFunctionManager.AddParticleTarget(owner, "Global_SS_Revive.troy", owner);
-            var visualBuff = ApiFunctionManager.AddBuffHudVisual("SummonerReviveSpeedBoost", 12.0f, 1, owner);
+            var visualBuff = ApiFunctionManager.AddBuffHudVisual("SummonerReviveSpeedBoost", 12.0f, 1, BuffType.COMBAT_ENCHANCER, owner, 12.0f);
             ApiFunctionManager.CreateTimer(12.0f, () =>
             {
-                ApiFunctionManager.RemoveBuffHudVisual(visualBuff);
```
Comment (MatthewFrench) [inline Champions/Global/SummonerRevive.cs]: Is there a downside to not removing the buff hud visual?

File: Champions/MasterYi/R.cs [hunk 2]
```diff
@@ -26,11 +27,11 @@ public void OnStartCasting(Champion owner, Spell spell, AttackableUnit target)
         {
             var p = ApiFunctionManager.AddParticleTarget(owner, "Highlander_buf.troy", target, 1);
             var buff = ((ObjAiBase) target).AddBuffGameScript("Highlander", "Highlander", spell);
-            var visualBuff = ApiFunctionManager.AddBuffHudVisual("Highlander", 10.0f, 1, owner);
+            var visualBuff = ApiFunctionManager.AddBuffHudVisual("Highlander", 10.0f, 1, BuffType.COMBAT_ENCHANCER,
+                owner, 10.0f);
             ApiFunctionManager.CreateTimer(10.0f, () =>
             {
                 ApiFunctionManager.RemoveParticle(p);
-                ApiFunctionManager.RemoveBuffHudVisual(visualBuff);
```
Comment (MatthewFrench) [inline Champions/MasterYi/R.cs]: Is there a downside to not removing the buff hud visual?

**Review by MatthewFrench — COMMENTED** [review]
Just a question about a few missing hud buff removals. Other than that looks good.

**Review by MatthewFrench — APPROVED** [review]

File: Champions/Global/SummonerRevive.cs (hunk 1 above)
Comment (yasinallahdev) [reply Champions/Global/SummonerRevive.cs]: The final argument in AddBuffHudVisual is a Timer Argument, which will cause the buff HUD visual to be automatically be removed after the set amount of time. This means that the CreateTimer() calls are (in several cases) unneeded extra code.

File: Champions/MasterYi/R.cs (hunk 2 above)
Comment (yasinallahdev) [reply Champions/MasterYi/R.cs]: See above comment regarding the same question in regards to SummonerRevive. I should probably use AddBuffHudVisual()'s Timer argument in all applicable cases for completion, though.

**Pull request merged**